### ISSUE TITLE
Lambda AddPermissions returning 404 Not Found

### ISSUE THREAD

**Issue by lucianthorr** [localstack/localstack#1260]
I'm trying to setup an lambda function triggered by S3 events within a localstack docker-compose environment.
I've got the latest localstack image running in one service with S3, Lambda and APIGATEWAY and the second service is using `awslocal` to communicate with the localstack.

I have gotten as far as successfully creating a lambda function but when I attempt to `add-permission` I get a 404 Not Found error.

My command is:
```
awslocal lambda add-permission 
--function-name my-lambda-function 
--action lambda:InvokeFunction 
--statement-id s3 
--principal s3.amazonaws.com 
--source-arn arn:aws:s3:::event-source-bucket
```

This works against AWS so I'm not sure what I need to have setup within localstack to get it working.  Thanks for any insight you may have.

**Comment by mralanlee** [comment]
Also getting the same issue.

**Comment by OrlandoHVProg** [comment]
+1. Any solution?

**Comment by HowWeRollingham** [comment]
+1

**Comment by TheCynosure** [comment]
+1

**Comment by anthonydelarosa** [comment]
so this is still an issue? anyone out there?

**Comment by danabrams** [comment]
+1

**Comment by CharlesConner** [comment]
I'm getting the same error

**Comment by kevin860** [comment]
I've also run into this issue. Turning on debug logging points to a 404 when retrieving the IAM policy:

`DEBUG - http://localhost:4574 "GET /2015-03-31/functions/MyApi/policy HTTP/1.1" 404 232`

I was able to reproduce the same 404 error by running `awslocal lambda get-policy --function-name MyApi --debug`.

For the record, the function is definitely there and I can access it by running `awslocal lambda get-function --function-name MyApi`

I haven't found any workarounds but maybe this can help point someone in the right direction?

**Comment by sunnylbk** [comment]
+1

**Comment by fionic** [comment]
+1

**Comment by whummer** [comment]
Thanks for reporting - this should be fixed in #1691 . @sunnylbk @kevin860 @CharlesConner @danabrams @anthonydelarosa @TheCynosure @HowWeRollingham @OrlandoHVProg @mralanlee @lucianthorr Please re-open this issue if the problem persists.

**Comment by fionic** [comment]
@whummer is there somewhere to track when this change will get pushed to the latest docker image?
